unicodetools/src/main/java/org/unicode/text/UCA/UCA.java

```diff
@@ -199,7 +199,7 @@ public void remove() {
         }
     }
 
-    private final class PrimaryIterable implements Iterable<UCA.Primary> {
+    public final class PrimaryIterable implements Iterable<UCA.Primary> {
```

Comment:
why public if we didn't need that before?

Reply:
I think this was to work around a JDK issue.

Reply:
> [ERROR] /Users/srl295/src/unicodetools/unicodetools/src/main/java/org/unicode/text/UCA/WriteCollationData.java:[1805,74] org.unicode.text.UCA.UCA.PrimaryIterable.iterator() is defined in an inaccessible class or interface

Reply:
yes. getIgnorableAndRegularPrimaries() returns a PrimaryIterable, and WriteCollationData tries to iterate over it. I hate to ask but how did this work before?  Maybe `getIgnorableAndRegularPrimaries()` should return the interface `Iterable<UCA.Primary>` if the class was meant to be kept private?

Reply:
Well, it did work this summer when I generated UCA data files several times.
I wonder if a later JDK got stricter? (Since you are upgrading to 11 I think.)

If the call sites don't need to know the specific impl class, then returning `Iterable<UCA.Primary>` would be fine.